lib/phoenix/controller/pipeline.ex

```diff
@@ -23,10 +23,25 @@ defmodule Phoenix.Controller.Pipeline do
                  &(&1 |> Map.put(:phoenix_controller, __MODULE__)
                       |> Map.put(:phoenix_action, action))
 
-        Phoenix.Endpoint.instrument conn, :phoenix_controller_call,
+        start = System.monotonic_time()
+        :telemetry.execute([:phoenix, :controller, :call, :start], %{time: System.system_time()}, %{
+          conn: conn,
+          controller: __MODULE__,
+          action: action
+        })
+        conn = Phoenix.Endpoint.instrument conn, :phoenix_controller_call,
           %{conn: conn, log_level: @phoenix_log_level}, fn ->
           phoenix_controller_pipeline(conn, action)
         end
+        stop = System.monotonic_time()
+        :telemetry.execute([:phoenix, :controller, :call, :stop], %{
```

Comment:
@binaryseed would you use the controller call stop for anything? This time is most likely not relevant since we already measure the whole plug pipeline.

Reply:
As long as I can get access to the `conn` from the plug `:stop` event (to get info like `conn.status`), I don't think this is strictly necessary

Will the Phoenix pipeline always contain `Plug.Telemetry`? What will it's event prefix be?

Reply:
@binaryseed my plan is to plug it with `[:phoenix, :endpoint]` and that will always give you when the response is set.

Reply:
@josevalim now I can't tell if I need the `stop` or not, hehe. What does "plug it with [:phoenix, :endpoint]" mean?

If spans are to be created in an event that is triggered at the start of a controller then there needs to be an event to finish the span when it finishes.

Reply:
@tsloughter we will emit events for the whole plug pipeline. That would be one span. We will also emit an event for the controller, but we were planning to not create a stop. But if you want a span, then we need the stop. I guess we will include the stop because it at leasts helps differenciate between events happening in controllers compared to other parts of the pipeline.

With that said, this PR is then good to be merged as is. :+1:

Reply:
Ah. Yea, the issue with a whole plug pipline start/stop is not being able to use the route as the name of the span, since that isn't matched yet, right?

Reply:
Correct. The router comes quite late in the pipeline.

The other idea is to dispatch this event in the router as soon as the route is known. The difference though is that any plug in the router will then also be included as part of the measurements. This can be positive, as business logic starts to be invoked already in the router in many cases.

@chrismccord, any preferences?

Reply:
Ok. It sounds like it will be similar to what we already do with `opencensus_plug` and `opencensus_phoenix`, but better since the plug part won't have to be manually put at the beginning of the sequence.

Basically a user enabling just the phoenix controller trace gets a span named for the route only covering the controller. But a user could also enable the trace to start from the initial plug handling, making the controller span a child of that initial span. This would allow for spans to be created as children in any of the plugs but the user has to either be ok with the path being used for the span name or defining a function that converts it to the span name.